osutil/user.go

```diff
@@ -305,12 +305,38 @@ func RealUser() (*user.User, error) {
 	}
 
 	real, err := user.Lookup(realName)
-	// can happen when sudo is used to enter a chroot (e.g. pbuilder)
-	if _, ok := err.(user.UnknownUserError); ok {
-		return cur, nil
-	}
+	// can happen when sudo is used to enter a chroot (e.g. lpbuilder)
 	if err != nil {
-		return nil, err
+		// Treat all non-nil errors as user.Unknown{User,Group}Error's, as
```

Comment:
as discussed, I will undo the RealUser changes here and instead add a comment and also rename the function to make it more clear that this function isn't checking for the "real user id" in the sense of effective vs real uids and is more just "if sudo give me the SUDO_USER, otherwise normal user"

Reply:
I renamed `RealUser` to `UserMaybeSudoUser`, probably not the best name, but less confusing than it was previously I think